modules/apps/product-navigation/product-navigation-taglib/src/main/resources/META-INF/resources/personal_menu/js/PersonalMenu.es.js

```diff
@@ -8,33 +8,33 @@ import {ClayDropDownWithItems} from '@clayui/drop-down';
 import ClayIcon from '@clayui/icon';
 import ClayLoadingIndicator from '@clayui/loading-indicator';
 import ClaySticker from '@clayui/sticker';
-import {fetch, navigate, openSelectionModal, sub} from 'frontend-js-web';
+import {fetch, getRenderer, sub} from 'frontend-js-web';
 import PropTypes from 'prop-types';
 import React, {useEffect, useRef, useState} from 'react';
 
 function mapItemsOnClick(items) {
 	return items.map((item) => {
-		const {items: nestedItems, jsOnClickConfig, ...otherKeys} = item;
+		const {
+			items: nestedItems,
+			jsOnClickConfig,
+			onClickJSModuleURL,
+			...otherKeys
+		} = item;
 
 		const newVal = {...otherKeys};
 
 		if (nestedItems) {
 			newVal.items = mapItemsOnClick(nestedItems);
 		}
 
-		if (jsOnClickConfig) {
-			newVal.onClick = () => {
-				const {selectEventName, title, url} = jsOnClickConfig;
-
-				openSelectionModal({
-					id: selectEventName,
-					onSelect(selectedItem) {
-						navigate(selectedItem.url);
-					},
-					selectEventName,
-					title,
-					url,
+		if (onClickJSModuleURL) {
+			newVal.onClick = async () => {
+				const {component} = await getRenderer({
```

Comment:
`getRenderer` is generally for client-extensions and not sure it really works for this use case, at least semantically because you aren't rendering anything, you are just executing a function and additionally you aren't calling a `component`.

I think what you really should be calling is just a simple `Liferay.Loader.require(url, ...`

Reply:
```js
		Liferay.Loader.require(
			onClickJSModuleURL,
			(jsModule) => jsModule.default(jsOnClickConfig) // not sure if you need the `.default` or not
		);
```

you may also need to wrap it in a promise